src/WorkflowCore/Services/ErrorHandlers/CatchHandler.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using WorkflowCore.Interface;
+using WorkflowCore.Models;
+using WorkflowCore.Models.LifeCycleEvents;
+
+namespace WorkflowCore.Services.ErrorHandlers
+{
+    public class CatchHandler : IWorkflowErrorHandler
```

Comment:
Could you implement a set of unit tests for this class that make it's behavior explicit?

Reply:
Sure, will do it soon. By the way I have a question concerning behaviour of `CompensateHandler`. I don't understand what lines 41-45 of `CatchHandler` do as I copypasted them from lines 56-60 of `CompensateHandler`, hence I don't understand if they are needed at all. Could you please elaborate on this?

Reply:
I believe that was for bubbling up the call stack of nested containers

Reply:
I clearly have troubles understanding this part of bubbling up the call stack. Could you please go into deeper details? And please tell me if it's needed or not in case of `Catch`clause.